### DISCUSSION TITLE
Use Filebeat module in SO 2.4

### DISCUSSION THREAD

**Discussion by a3hiennt** [Security-Onion-Solutions/securityonion#10722] [2.4] [answered]
Dear SO teams.
We using Filebeat module in our SO 2.3 to parse our log such as Fortinet, Juniper, ASA devices.
https://www.elastic.co/guide/en/beats/filebeat/current/filebeat-modules.html
Can you help me how to use this function on SO 2.4?
Regards.

**Comment by dougburks** [comment]
In 2.4, you will want to use Elastic Agent instead of Filebeat:
https://docs.securityonion.net/en/2.4/elastic-agent.html

**Reply by jdonovan-itcnp** [reply]
@dougburks Just to confirm, this means that 2.4 will not be compatible with Beats at all, correct?

Is it fair to assume that there is a planned path to transition between 2.3 and 2.4 without cutting off existing data sources?

**Reply by dougburks** [reply]
@jdonovan1013 You may be able to make Beats work with 2.4, but our officially supported recommendation is Elastic Agent. Under the hood, Elastic Agent runs several existing Beats so you should have coverage for your existing data sources and then some. If you have further questions, please start a separate discussion rather than replying to somebody else's discussion.

**Comment by a3hiennt** [comment, marked as answer]
I find out this information.
We can use Integrations in Elastic Agent instead of Module in Filebeat. 
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/80515844/a0f2ac8b-351b-4e03-81e9-266821d878cc)
https://www.elastic.co/guide/en/fleet/current/beats-agent-comparison.html
Thanks @dougburks
